src/js/providers/flash.js

```diff
@@ -206,13 +222,13 @@ define([
 
                     // The browser may block the flash object until user enables it
                     var _this = this;
-                    _flashBlockedTimeout = setTimeout(function() {
-                        Events.trigger.call(_this, 'flashBlocked');
-                    }, 4000);
-                    _swf.once('embedded', function() {
-                        clearTimeout(_flashBlockedTimeout);
-                        Events.trigger.call(_this, 'flashUnblocked');
-                    }, this);
+
+                    // Wait until the window gets focus to see check flash is blocked
+                    if (document.hasFocus()) {
+                        checkFlashBlocked();
+                    } else {
+                        window.addEventListener('focus', onFocus);
```

Comment:
@pajong you have to be more careful when adding event listeners that you remove them more aggressively. Here's some scenarios you need to cover:

- This provider is removed
- Flash successfully sets up in a background tab (you need to test the same steps with "Run all plugin content" selected). This means on swf 'ready' or anywhere we also call `clearTimeout(_flashBlockedTimeout)`